### DISCUSSION TITLE
buy with stop parameter buys far under the current price

### DISCUSSION THREAD

**Discussion by Hungryarie** [kernc/backtesting.py#403] [General]
Hi, thanks for making this wonderfull programm!
I've made a strategy and I am playing with the parameters, but when I place a buy order with the stop parameter it will buy far below the actual price.

curent OHLC: O:9664.45, H;9728.13, L;9661.0, C:9717.73

`self.buy(stop=8940.3, limit=9717.7, tp=11272.6)`
the trade is done with EntryPrice: 8958.19 ExitPrice: 9877.31
![buy-stop](https://user-images.githubusercontent.com/684364/124279669-10263600-db48-11eb-8b50-0b205a46b5a3.png)

You can see in the image that the trade lines are also off.

but when I use
`self.buy(sl=8940.3, limit=9717.7, tp=11272.6)`
the trade is down with EntryPrice: 9737,155 Exitprice: 9877.33, just as you would expect.
![buy-stoploss](https://user-images.githubusercontent.com/684364/124279705-1a483480-db48-11eb-8ed4-4153bd413311.png)

Am I using the stop parameter wrong?

thanks!

**Comment by kernc** [comment]
This is [`Backtest(..., commission=)`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest) at work, adjusting the entry price.

**Reply by Hungryarie** [reply]
But why the difference between the two options? (Stop vs stoploss)

**Reply by CadeHalcyon** [reply]
Just chipping in - stop price and stop loss are two different things entirely.  Recommend reading up on them both.

**Reply by Hungryarie** [reply]
thanks, so a stop parameter is in fact a pseudo buy parameter at a fixed price.
But how can the algo think is has been bough if the price never reached this stopprice in the example above?

**Reply by kernc** [reply]
When the stop price is hit, the order becomes either a [market order (`limit=0`)](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Order.limit) or a [stop-limit order (`limit != 0`)](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Order.limit).

> But how can the algo think is has been bough if the price never reached this stopprice in the example above?

Please see my answer above. :stuck_out_tongue:
